### ISSUE TITLE
Bug: invalid sum of terms of the phi formula in approximateEntropy.c

### ISSUE THREAD

**Issue by zasderjjrr123** [arcetri/sts#7]
While running the suite, I decided to test a run of 2 ^ 20 0 bits, to confirm the library would fail all tests. To my surprise, it still passed the approximate entropy test.

After some digging in the code, I found that the `compute_phi` function in `approximateEntropy.c` uses the `log` function on all members of the array `state->apen_C[thread_state->thread_id]`. If any of these members are zero, it returns as `nan`, which messes up future calculations until the test gives a false pass.

Bug:
```
         /*
         * Step 3 and 4a: compute the the terms of the phi formula
         */
        sum = 0.0;
        for (i = 0; i < powLen; i++) {
                sum += (double) state->apen_C[thread_state->thread_id][i] *
                                log(state->apen_C[thread_state->thread_id][i] / (double) n);
        }

```

My fix:
```
        /*
         * Step 3 and 4a: compute the the terms of the phi formula
         */
        sum = 0.0;
        for (i = 0; i < powLen; i++) {
                if (state->apen_C[thread_state->thread_id][i]) {
                        sum += (double) state->apen_C[thread_state->thread_id][i] *
                                log(state->apen_C[thread_state->thread_id][i] / (double) n);
                }
        }
```

**Comment by lcn2** [comment]
At first glance, @zasderjjrr123 you fix appears to be correct.

We will look at to some more in the next few days.

Thank you!

**Comment by lcn2** [comment]
Thank you @zasderjjrr123 fixed in 60af0756ad71a4b99ab5d75428b5c88f31e72a96 

See also commit 060abcba86bdfd93cbe458a3d37a1a1f217c70f2
